src/core.ts

```diff
@@ -60,11 +62,14 @@ export function teardown() {
   }
 
   delete document[ClarityAttribute];
+  if (compressionWorker) {
+    compressionWorker.terminate();
+  }
   state = State.Unloaded;
 
-  // Upload residual events
+  // Instrument teardown and upload residual events
   instrument({ type: Instrumentation.Teardown });
-  uploadNextPayload();
+  uploadPendingEvents();
```

Comment:
>uploadPendingEvents [](start = 2, length = 19)

How does compression happen for these if you terminate the worker above?

Reply:
We compress this in the main thread for teardown (for example on 'unload') on purpose

---
In reply to: [137239714](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137239714) [](ancestors = 137239714)